File: backend/domain/table/payment.go
package table

import (
	"time"

	z "github.com/Oudwins/zog"
)

type Payment struct {
	ID                string         `json:"id"`
	UserID            int            `json:"userId"`
	TableID           int            `json:"tableId"`
	Variants          []OrderVariant `json:"variants"`
	TotalPaymentCents int            `json:"totalPaymentCents"`
	Comment           string         `json:"comment"`
	RegisteredAt      time.Time      `json:"registeredAt"`
}

var paymentSchema = z.Struct(z.Shape{
	"ID":                z.String().UUID().Required(),
	"UserID":            z.Int().GTE(1).Required(),
	"TableID":           z.Int().GTE(1).Required(),
	"Variants":          z.Slice(orderVariantSchema).Min(1).Required(),
	"TotalPaymentCents": z.Int().GTE(0).Required(),
	"Comment":           z.String().Max(100),
	"RegisteredAt":      z.Time().Required(),
})

File: frontend/src/service/table/Payment.ts
import { z } from 'zod'

import { LineItemSchema } from './Order'

export const PaymentSchema = z.object({
  id: z.uuid(),
  userId: z.number().int().min(1),
  tableId: z.number().int().min(1),
  variants: LineItemSchema.array().min(1),
  totalPaymentCents: z.number().int().min(0),
  comment: z.string().max(100),
  registeredAt: z.string().refine((date) => !isNaN(Date.parse(date)), {
    message: 'Invalid date format',
  }),
})
export type Payment = z.infer<typeof PaymentSchema>

export const RegisterPaymentSchema = z.object({
  tableId: z.number().int().min(1),
  variants: LineItemSchema.array().min(1),
  comment: z.string().max(100),
})
export type RegisterPayment = z.infer<typeof RegisterPaymentSchema>
